corona-warn-app/cwa-app-android
# Add app shortcut for contact diary (EXPOSUREAPP-4659) (closes #2130)

### Description

As written in #2130 an app shortcut for the contact diary could improve the user experience quickly navigating there. This PR adds such a shortcut. The `Intent` of the shortcut points to the `LauncherActivity` to make sure the existing flow does not get broken. I modified the conditions a little bit but all in all it should be similar like before:

1. Check, if an update is needed and possibly show the update dialog
2. Check, if the user is not onboarded yet and possibly show the onboard dialog
3. Otherwise fire the renamed event `GoToAppShortcutOrMainActivity` which will tell the `LauncherActivity` to evaluate if an app shortcut should be opened or we finally open the `MainActivity` based on the intent extras.

### Screenshots

Long press on app launcher icon:

![2130-homescreen](https://user-images.githubusercontent.com/10376534/104940536-2aad8e00-59b2-11eb-9ad7-3571e3230a1b.png)

After clicking the contact diary shortcut:

![2130-after-shortcut](https://user-images.githubusercontent.com/10376534/104940545-2c775180-59b2-11eb-8c35-3f8f7f580d1b.png)

### How to test

- Long press on app launcher icon
- Select `Kontakt-Tagebuch` (or `Kontakt-Tagebuch öffnen` if long description is shown)

### TODOs

- Please check `shortcutShortLabel` and `shortcutLongLabel`, especially their translations. I'm not familiar with how your translation system works and how to trigger a needed translation from the team. I added german strings (they need to be checked too) as well as strings for the default but with empty content.
- I used the `ic_diary_book` for the icon. The android documentation says app shortcut icons should be 24x24dp which is not the case for the used image. It seems to work but should definitely be changed with a smaller icon.
- Due to the flow described above there is an edge case: When the user wasn't onboarded yet, the onboarding flow will be shown and afterwards the `MainActivity`. I'm not sure if this is a problem from a UX perspective. If yes we could maybe disable the shortcut and enable it after the user was onboarded.

Feel free to tackle any of this todos or ping me with the provided information and I will do so 😄


---
Internal Tracking ID: [EXPOSUREAPP-4659](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-4659)
